lib/logstash/outputs/logstash.rb

```diff
@@ -169,7 +169,7 @@ def normalize_host_uris
   end
 
   def send_events(events)
-    body = LogStash::Json.dump(events.map(&:to_hash))
+    body = events.map(&:to_json).map { |json| json << "\n" }.join
```

Comment:
To avoid the extra final string we create with [].join can we modify instead the gzip method to take an array of strings and iterate through it instead?

Reply:
We _could_ but the uncompressed body is currently passed through to `#transmit` (for logging purposes), and I was trying to avoid changing too much.

If we're intent on saving that extra object before the others are immediately eligible for GC, we could rely on a reduction, which would mutate the _first_ instead of creating a new object:

```suggestion
    body = events.map(&:to_json).map { |json| json << "\n" }.reduce(:<<)
```

And if we'd like things to fall out of scope quicker, we could make the enum lazy:

```suggestion
    body = events.lazy.map(&:to_json).map { |json| json << "\n" }.reduce(:<<)
```

Reply:
Given we're changing how the string that is sent to be gziped is formed, IMO it's within scope to go a reasonable length to avoid waste given how much this code is in the critical path for a Logstash to Logstash pipeline.

I like the `.reduce(:<<)` idea. +1 on going with it.